database.sql

```diff
@@ -0,0 +1,116 @@
+CREATE TABLE brands (
+  id bigserial,
+  name text NOT NULL UNIQUE,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE products (
+  id bigserial,
+  brand_id bigint NOT NULL,
+  category_id bigint NOT NULL,
+  name text NOT NULL UNIQUE,
+  price money NOT NULL,
+  quantity bigint NOT NULL,
+  created_at timestamp NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (brand_id) REFERENCES brands (id) ON DELETE CASCADE
+);
+
+CREATE TYPE user_type AS ENUM ('admin', 'customer');
+
+CREATE TABLE users (
+  id bigserial,
+  user_type user_type NOT NULL,
+  email text NOT NULL UNIQUE,
+  password text NOT NULL,
+  phone text,
+  first_name text,
+  last_name text,
+  created_at timestamp NOT NULL,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE addresses (
+  id bigserial,
+  user_id bigint NOT NULL,
+  address text NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (user_id) REFERENCES users (id) ON DELETE CASCADE
+);
+
+CREATE TYPE order_status_type AS ENUM ('inreview', 'indelivery', 'completed');
+
+CREATE TABLE orders (
+  id bigserial,
+  user_id bigint NOT NULL,
+  price money NOT NULL,
+  address_id bigint NOT NULL,
+  created_at timestamp NOT NULL,
+  order_status order_status_type NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (user_id) REFERENCES users (id) ON DELETE RESTRICT,
+  FOREIGN KEY (address_id) REFERENCES addresses (id) ON DELETE RESTRICT
+);
+
+CREATE TABLE orders_items (
+  product_id bigint,
+  order_id bigint,
+  quantity bigint NOT NULL,
+  PRIMARY KEY (product_id, order_id),
+  FOREIGN KEY (product_id) REFERENCES products (id) ON DELETE RESTRICT,
+  FOREIGN KEY (order_id) REFERENCES orders (id) ON DELETE CASCADE
+);
```

Comment:
Should the price field also be added to the orders_items table (as i see on the other market platforms it is a common case then you can check a price of an every item you ordered)?

Reply:
It makes sence